paw-webapp/service/src/main/java/ar/edu/itba/paw/webapp/service/MailServiceImpl.java

```diff
@@ -37,57 +37,60 @@ private InternetAddress from() {
             return null;
         }
     }
 
     @Async
-    public void sendPasswordResetEmail(User user, String newPassword) {
+    public void sendPasswordResetEmail(User user, String url) {
         MimeMessagePreparator preparator = new MimeMessagePreparator() {
             @Override
             public void prepare(MimeMessage mimeMessage) throws Exception {
                 mimeMessage.setFrom(from());
                 mimeMessage.setRecipient(Message.RecipientType.TO,
                         new InternetAddress(user.getEmail()));
-                mimeMessage.setText("Dear "
-                        + user.getUsername()
-                        + ", \nYour new password is: "
-                        + newPassword
-                        + "\nPlease log in with your new password, and if you wish, head to your profile page and change your password to something easier to remember.");
+                mimeMessage.setText("Dear " + user.getUsername() + ", \n" +
+                        "This email was sent to you because someone requested a password reset on your account. \n\n" +
+                        "Visit the following link to set a new password: \n\n" +
+                        url + "\n\n\n" +
+                        "Yours sincerely, \n" +
+                        "PowerUp team.");
                 mimeMessage.setSubject("PowerUp password reset");
             }
         };
 
         try {
             mailSender.send(preparator);
-            logger.info("Sent new password email to {} ({})", user.getUsername(), user.getEmail());
+            LOGGER.info("Sent reset password email to {} ({})", user.getUsername(), user.getEmail());
         } catch (MailException ex) {
-            logger.error("Couldn't send new password email to {} ({}): {}", user.getUsername(), user.getEmail(), ex);
+            LOGGER.error("Couldn't send reset password email to {} ({}): {}", user.getUsername(), user.getEmail(), ex);
         }
     }
 
     @Override
+    @Async
     public void sendPasswordChangedEmail(User user) {
         MimeMessagePreparator preparator = new MimeMessagePreparator() {
             @Override
             public void prepare(MimeMessage mimeMessage) throws Exception {
                 mimeMessage.setFrom(from());
                 mimeMessage.setRecipient(Message.RecipientType.TO,
                         new InternetAddress(user.getEmail()));
-                mimeMessage.setText("Dear "
-                        + user.getUsername()
-                        + ", \nYour password was changed. If you haven't changed your password, please reset your password at"
-                        + " the Log In page clicking on \"Forgot Password?\"");
+                mimeMessage.setText("Dear " + user.getUsername() + ", \n" +
+                        "Your password was changed.\n" +
+                        "If it wasn't you, please reset it clicking on \"Forgot Password?\" in the log in page.\n\n" +
```

Comment:
Agregá `Or if you're already logged in, by clicking on the "Change Password" button in your profile page.`